### PR TITLE
feat: sss in case of err on creation, servers should be recreated

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
### Description of your changes

This PR adds:

## Checklist

I have:

- [ ] Add PR name as appropriate (e.g. `feat`/`fix`/`doc`/`test`/`refactor`)
- [ ] Run `make reviewable` and `make crds.clean` to ensure the PR is ready for review
- [ ] Add or update tests (if applicable)
- [ ] Add or update Documentation using `make docs.update` (if applicable)
- [ ] Update `docs/CHANGELOG.md` file (label: `upcoming release`)
- [ ] Check Sonar Cloud Scan
- [ ] Update Github or Jira Issue

### CODE REVIEW

File: internal/controller/compute/statefulserverset/serverset_controller.go [hunk 1]
```diff
@@ -148,3 +166,32 @@ func extractSSetFromSSSet(sSSet *v1alpha1.StatefulServerSet) *v1alpha1.ServerSet
 func getSSetName(cr *v1alpha1.StatefulServerSet) string {
 	return fmt.Sprintf("%s-%s", cr.Name, cr.Spec.ForProvider.Template.Metadata.Name)
 }
+
+// Delete - deletes the server k8s client and waits until it is deleted
+func (k *kubeServerSetController) Delete(ctx context.Context, name, namespace string) error {
```
Comment (cbeti-ionos) [inline internal/controller/compute/statefulserverset/serverset_controller.go:171]: Should this also be added to the `kubeSSetControlManager` interface?

File: internal/controller/compute/statefulserverset/serverset_controller.go [hunk 2]
```diff
@@ -148,3 +166,32 @@ func extractSSetFromSSSet(sSSet *v1alpha1.StatefulServerSet) *v1alpha1.ServerSet
...
+	if err != nil {
+		return err
+	}
+	if err := k.kube.Delete(ctx, condemnedServer); err != nil {
+		return fmt.Errorf("error deleting server %w", err)
+	}
+	return kube.WaitForResource(ctx, kube.ResourceReadyTimeout, k.isServerDeleted, condemnedServer.Name, namespace)
+}
+
+func (k *kubeServerSetController) isServerDeleted(ctx context.Context, name, namespace string) (bool, error) {
```
Comment (cbeti-ionos) [inline internal/controller/compute/statefulserverset/serverset_controller.go:182]: isServerSetDeleted

File: internal/controller/compute/statefulserverset/serverset_controller.go [hunk 3]
```diff
@@ -148,3 +166,32 @@ func extractSSetFromSSSet(sSSet *v1alpha1.StatefulServerSet) *v1alpha1.ServerSet
...
+	k.log.Info("Checking if Serverset is deleted", "name", name, "namespace", namespace)
+	obj := &v1alpha1.Server{}
```
Comment (cbeti-ionos) [inline internal/controller/compute/statefulserverset/serverset_controller.go:184]: &v1alpha1.ServerSet{}

File: internal/controller/compute/statefulserverset/serverset_controller.go [hunk 4]
```diff
@@ -148,3 +166,32 @@ func extractSSetFromSSSet(sSSet *v1alpha1.StatefulServerSet) *v1alpha1.ServerSet
 func getSSetName(cr *v1alpha1.StatefulServerSet) string {
 	return fmt.Sprintf("%s-%s", cr.Name, cr.Spec.ForProvider.Template.Metadata.Name)
 }
+
+// Delete - deletes the server k8s client and waits until it is deleted
+func (k *kubeServerSetController) Delete(ctx context.Context, name, namespace string) error {
+	condemnedServer, err := k.Get(ctx, name, namespace)
```
Comment (cbeti-ionos) [inline internal/controller/compute/statefulserverset/serverset_controller.go:172]: not sure if `condemned` naming also applies to ServerSet here

File: internal/controller/compute/statefulserverset/statefulserverset.go [hunk 5]
```diff
@@ -109,55 +108,90 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
...
+func (e *external) isServerSetUpToDate(ctx context.Context, cr *v1alpha1.StatefulServerSet) (creationServerUpToDate bool, serverUpToDate bool, err error) {
+	_, err = e.SSetController.Get(ctx, getSSetName(cr), cr.Namespace)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			return false, false, nil
+		}
+	}
+	serverUpToDate, err = areSSetResourcesUpToDate(ctx, e.kube, cr)
+	if err != nil {
+		return true, true, err
```
Comment (cbeti-ionos) [inline internal/controller/compute/statefulserverset/statefulserverset.go:178]: is `true` here a typo?

File: internal/controller/compute/statefulserverset/serverset_controller.go (hunk 3 above)
Comment (cristiGuranIonos) [reply internal/controller/compute/statefulserverset/serverset_controller.go:184]: this actually explains a bug I had while testing. thanks!

File: internal/controller/compute/statefulserverset/statefulserverset.go (hunk 5 above)
Comment (cristiGuranIonos) [reply internal/controller/compute/statefulserverset/statefulserverset.go:178]: it basically doesn't matter as the error will prevent anything from happening

File: internal/controller/compute/statefulserverset/serverset_controller.go (hunk 1 above)
Comment (cristiGuranIonos) [reply internal/controller/compute/statefulserverset/serverset_controller.go:170]: Not used externally, so not required. Might be better to be lowercased

**Review by cbeti-ionos — APPROVED** [review]
:+1:

**Pull request merged**